PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/KustoQueriesWithTimeIntervalInQueryStringCheck.java

```diff
@@ -0,0 +1,255 @@
+         * If the element is a PsiLocalVariable, the method checks the initializer of the variable
+         * If the element is a PsiPolyadicExpression, the method processes the expression to replace the variables with their values
+         * The method then checks the expression for the anti-patterns by matching regex patterns with the expression text
+         * and registers a problem if an anti-pattern is detected
+         *
+         * @param element - the element to visit
+         */
+        @Override
+        public void visitElement(@NotNull PsiElement element) {
+            super.visitElement(element);
+
+            // Skip the whole rule if the rule configuration is empty
+            if (SKIP_WHOLE_RULE) {
+                return;
+            }
+
+            if (!(element instanceof PsiPolyadicExpression || element instanceof PsiLocalVariable || element instanceof PsiMethodCallExpressionImpl)) {
+                return;
+            }
+
+            if (element instanceof PsiMethodCallExpressionImpl) {
+                checkParameterNames((PsiMethodCallExpressionImpl) element);
+            }
+
+            if (element instanceof PsiLocalVariable) {
+                PsiLocalVariable variable = (PsiLocalVariable) element;
+                handleLocalVariable(variable);
+            }
+
+            // if element is a polyadic expression, extract the value and replace the variable with the value
+            // PsiPolyadicExpressions are used to represent expressions with multiple operands
+            // eg ("datetime" + startDate), where startDate is a variable
+            if (element instanceof PsiPolyadicExpression) {
+                handlePolyadicExpression((PsiPolyadicExpression) element);
+            }
+        }
+
+        /**
+         * This method checks the expression for the anti-patterns by matching regex patterns with the expression text
+         * and registers a problem if an anti-pattern is detected
+         *
+         * @param expression - the expression to check
+         * @param element    - the element to check
+         */
+        void checkExpression(PsiExpression expression, PsiElement element) {
+            if (expression == null) {
+                return;
+            }
+            String text = expression.getText();
+
+            // Check if the expression text contains any of the regex patterns
+            boolean foundAntiPattern = REGEX_PATTERNS.stream().anyMatch(pattern -> pattern.matcher(text).find());
+
+
+            // If an anti-pattern is detected, register a problem
+            if (foundAntiPattern) {
+                PsiElement parentElement = element.getParent();
+
+                if (parentElement instanceof PsiLocalVariable) {
+                    PsiLocalVariable variable = (PsiLocalVariable) parentElement;
+                    String variableName = variable.getName();
+                    timeIntervalParameters.add(variableName);
```

Comment:
If the user fixes the problem, where do clear this list?

Reply:
if the user fixes the problem the visitor will parse the source code again with a newly initialised empty list -- the list is now cleared before every visit